iotedgedev/modules.py

```diff
@@ -171,57 +172,58 @@ def build_push(self, template_file, default_platform, no_build=False, no_push=Fa
-                if not no_build:
-                    build_profile = tag_build_profile_map.get(tag)
-
-                    module_name = build_profile.module_name
-                    dockerfile = build_profile.dockerfile
-                    self.output.info("BUILDING MODULE: {0}".format(module_name))
-                    self.output.info("PROCESSING DOCKERFILE: {0}".format(dockerfile))
-                    self.output.info("BUILDING DOCKER IMAGE: {0}".format(tag))
-
-                    build_options = build_profile.extra_options
-                    build_options_parser = BuildOptionsParser(build_options)
-                    sdk_options = build_options_parser.parse_build_options()
-
-                    context_path = build_profile.context_path
-
-                    # A hack to work around Python Docker SDK's bug with Linux container mode on Windows
-                    # https://github.com/docker/docker-py/issues/2127
-                    dockerfile_relative = os.path.relpath(dockerfile, context_path)
-                    if docker.get_os_type() == "linux" and sys.platform == "win32":
-                        dockerfile_relative = dockerfile_relative.replace("\\", "/")
-
-                    build_args = {"tag": tag, "path": context_path, "dockerfile": dockerfile_relative}
-                    build_args.update(sdk_options)
-
-                    response = docker.docker_api.build(**build_args)
-                    docker.process_api_response(response)
-                if not no_push:
-                    docker.init_registry()
-
-                    # PUSH TO CONTAINER REGISTRY
-                    self.output.info("PUSHING DOCKER IMAGE: " + tag)
-                    registry_key = None
-                    for key, registry in self.envvars.CONTAINER_REGISTRY_MAP.items():
-                        # Split the repository tag in the module.json (ex: Localhost:5000/filtermodule)
-                        if registry.server.lower() == tag.split('/')[0].lower():
-                            registry_key = key
-                            break
-                    if registry_key is None:
-                        self.output.info("Could not find registry credentials with name {0} in environment variable. Pushing anonymously.".format(tag.split('/')[0].lower()))
-                        response = docker.docker_client.images.push(repository=tag, stream=True)
-                    else:
-                        response = docker.docker_client.images.push(repository=tag, stream=True, auth_config={
-                            "username": self.envvars.CONTAINER_REGISTRY_MAP[registry_key].username,
-                            "password": self.envvars.CONTAINER_REGISTRY_MAP[registry_key].password})
-                    docker.process_api_response(response)
-            self.output.footer("BUILD COMPLETE", suppress=no_build)
-            self.output.footer("PUSH COMPLETE", suppress=no_push)
+        if not no_build or not no_push:
```

Comment:
so it will build even if user specify no_build but push?

Reply:
No. There are extra check in Line 180 and Line 206. Adding check in Line 175 just to avoid creating `Docker` object on Line 176 when users only want to genconfig.